packages/panna-sdk/src/react/hooks/use-buy-with-fiat-quotes.ts

```diff
@@ -0,0 +1,135 @@
+import { useQuery, UseQueryOptions } from '@tanstack/react-query';
+import { toWei } from 'thirdweb/utils';
+import { getOnrampProviders, onRampPrepare } from '../../core/onramp';
+import type { BuyWithFiatQuote } from '../types/buy-with-fiat-quote.types';
+import {
+  DEFAULT_STALE_TIME,
+  DEFAULT_REFETCH_INTERVAL,
+  DEFAULT_RETRY_DELAY,
+  createDefaultRetryFn
+} from './constants';
+import { usePanna } from './use-panna';
+
+type UseBuyWithFiatQuotesParams = {
+  countryCode: string;
+  tokenAddress?: string;
+  cryptoAmount?: string;
+  receiver?: string;
+};
+
+/**
+ * Hook to retrieve buy with fiat quotes for a specific country with pricing information
+ * @param params - Parameters for retrieving buy with fiat quotes
+ * @returns React Query result with buy with fiat quotes data
+ */
+export function useBuyWithFiatQuotes(
+  {
+    countryCode,
+    tokenAddress,
+    cryptoAmount,
+    receiver
+  }: UseBuyWithFiatQuotesParams,
+  options?: Omit<UseQueryOptions<BuyWithFiatQuote[]>, 'queryKey' | 'queryFn'>
+) {
+  const { client } = usePanna();
+  const hasValidCountryCode = Boolean(countryCode);
```

Comment:
This can be easily verified more accurately using the `getCountry` method from the `iso-3166-1-alpha-2` module.
Ref: https://github.com/LiskHQ/panna-sdk/blob/development/packages/panna-sdk/src/core/onramp/onramp.ts#L186-L188